### PR TITLE
feature :  add tablet layout.

### LINKED ISSUE #453: Add Master Detail layout for tablets
**Actual Behaviour**

Currently, Tablets or devices with large screen show stretched layout in landscape mode, which can create uneasiness in scanning lines of messages, which is an issue for a chat application

**Expected Behaviour**

A master detail layout should be shown in tablets or large devices so as to utilize the screen space more efficiently and make it a pleasant experience for the user

**Would you like to work on the issue?**

Yes, I would like to work on the issue

### PR DESCRIPTION
**Pull request by divs4debu**
Add tablet layout  showing navigation drawer all the time.

Fixes #453 .

**Summary of changes**

[Explain here what changes were made and why.]

Screenshots or a brief video showing the change in action:

Please make sure these boxes are checked before submitting your pull request - thanks! [Guide](./PULL_REQUEST_GUIDE.md)

- [x] Code is formatted.

- [x] Run the lint tests with `./gradlew lintDebug` and make sure BUILD is SUCCESSFUL

- [x] If new feature is implemeted then it is compatible with Night theme too.

- [x] Commit messages are well-written.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@divs4debu, thanks for the pull request! It looks like you haven't yet signed the Dropbox CLA. Please [sign it here](https://opensource.dropbox.com/cla/).

**Comment by divs4debu** [comment]
@Sam1301 @timabbott @kunall17 @saketkumar95  Please review .

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*
				
@divs4debu, thanks for signing the CLA!

File: app/src/main/res/layout-sw600dp/main.xml [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                    <RelativeLayout
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:layout_marginLeft="2dp"
+                        android:layout_marginRight="2dp"
+                        android:background="@android:color/white"
+                        android:paddingBottom="12dp"
+                        android:paddingLeft="4dp"
+                        android:paddingRight="4dp"
+                        android:paddingTop="12dp">
```
Comment (saketkumar95) [inline app/src/main/res/layout-sw600dp/main.xml]: Add these dimensions in dimens.xml file..

Comment (divs4debu) [inline app/src/main/res/layout-sw600dp/main.xml]: should i also add navigation drawer dimension also into dimens.xml

**Comment by divs4debu** [comment]
@saketkumar95 Please review

**Comment by divs4debu** [comment]
@Sam1301 sorry for delayed reply i was busy with my mid semesters . I see what u are saying ,i have idea , i searched net , we can have bools.xml separately for value-sw600dp and general and have isTablet item in both of them but in sw600dp isTablet is true and in general false. what do you think?

**Comment by Sam1301** [comment]
@divs4debu yeah that would work 👍

**Comment by divs4debu** [comment]
@Sam1301 @saketkumar95 
Please Review

**Comment by divs4debu** [comment]
@Sam1301 @saketkumar95 Please Review

**Comment by Sam1301** [comment]
@divs4debu this looks great! 
Can you also post some screenshots for the changes?

**Comment by divs4debu** [comment]
# Size 1





# Size 2





# Size 3

**Comment by Sam1301** [comment]
@divs4debu thanks for the screenshots 👍 
Faced one issue: `main.xml` didn't render properly in homeview on 720X1280 device.

Steps to reproduce:
* narrow to stream from homeview (tapping on the messageheader)
* narrow to any topic
* press back

Here's a short video https://drive.google.com/open?id=0B4jK5QX65b0ZWDA5aGxsbldjLUU

**Comment by divs4debu** [comment]
@Sam1301 Thanks for pointing that out . i looked at problem, what i tried is switched to my master branch and run the application.
I found the same thing happen.
Steps i took :
- open app
- open topic stream nav
- select android
- select message header
- click back 

here the video -
https://drive.google.com/open?id=0BzZnaFq4F3EdS0VfcXZsR3h6SG8
please look at this and guide me how to proceed.

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 2]
```diff
@@ -341,10 +341,14 @@ public void onClick(View v) {
         commonProgressDialog = new CommonProgressDialog(this);
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
-        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        if(!isTablet()) {
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Please format the code here!

File: app/src/main/res/layout/main.xml [hunk 3]
```diff
@@ -48,15 +48,15 @@
...
+            android:layout_margin="@dimen/margin_fab"
             android:src="@drawable/ic_mode_edit_24dp"
             app:backgroundTint="@color/colorPrimary"
             app:borderWidth="0dp"
             app:layout_behavior="com.zulip.android.util.RemoveViewsOnScroll" />
     </android.support.design.widget.CoordinatorLayout>
 
     <android.support.design.widget.NavigationView
-        android:layout_width="320dp"
+        android:layout_width="@dimen/navigation_drawer_width"
```
Comment (kunall17) [inline app/src/main/res/layout/main.xml]: 👍 for these changes

**Comment by kunall17** [comment]
I think for the portrait mode we can use the default layout (like on the phone, hidden stream drawer) so the tablet will be able to use the full width and get better messages view,
Whats your thought on this @Sam1301 @divs4debu ?

Mainly because as seen here, no point of having the stream drawer open all the time!
<a href="https://cloud.githubusercontent.com/assets/12700799/24452979/9026a2e2-14a3-11e7-87a4-362061571875.png" >Screenshot</a>

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java (hunk 2 above)
Comment (divs4debu) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: @kunall17 Please can you help me know what to format?

Comment (vishwesh3) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: `Ctrl + Alt + L`
http://stackoverflow.com/a/16580200/5612089

**Comment by divs4debu** [comment]
@kunall17 I understand what u suggested  i can do that by specifying my sw600dp layout just for  layout-land but before doing any modification i would like more people to come up with their opinion

**Comment by divs4debu** [comment]
@Sam1301 I thoroughly checked the master branch and the error is there too, meaning it is not related to my changes. What should I do?

**Comment by Sam1301** [comment]
@divs4debu yes, can you open an issue for it?
Looking at the pr now.

**Comment by Sam1301** [comment]
@kunall17 just noticed your earlier comment,
> I think for the portrait mode we can use the default layout (like on the phone, hidden stream drawer) so the tablet will be able to use the full width and get better messages view,

I agree completely with this approach.

**Comment by Sam1301** [comment]
We can however have sidebar open for 10'' tablets. For 7'' tablets in port view, it makes sense to close the stream sidebar. @divs4debu

**Comment by divs4debu** [comment]
@Sam1301 I understood . I am on it and i will open issue too .

**Comment by divs4debu** [comment]
@Sam1301 Which one should i prefer for 10inch tablet-
## PIC 1
10inch2-por

## PIC 2
10inch-port

**Comment by kunall17** [comment]
@divs4debu PIC 2 seems much better 👍 
its better to give the user an option to open/close the stream drawer! 

Also can you complete the changes soon, I want to include this feature in upcoming beta release which we we'll be doing by the weekend!

**Comment by divs4debu** [comment]
@kunall17 yes I will complete this feature as fast as possible.

**Comment by divs4debu** [comment]
@kunall17  @Sam1301 
I have done most of the work but to achieve the feature i have to move most of the code in ZulipAcitivity's onCreate to the new function and then call that function in onConfigurationchanged. is that ok?

**Comment by divs4debu** [comment]
@kunall17  @Sam1301 
I have done most of the work but to achieve the feature i have to move most of the code in ZulipAcitivity's onCreate to the new function and then call that function in onConfigurationchanged. is that ok?

**Comment by kunall17** [comment]
Send the code in another commit, it would be easier to review!
You must be using onConfigurationchanged for the orientation change right?

**Comment by divs4debu** [comment]
@kunall17 I have added new commit please review it as soon as possible so that i could help you to  complete your target to release beta by this weekend.

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 4]
```diff
@@ -2242,6 +2248,7 @@ public void onNarrowFillSendBoxStream(String stream, String subject, boolean ope
 
     public void onNarrow(NarrowFilter narrowFilter) {
         if (narrowedList == null || !narrowedList.filter.equals(narrowFilter)) {
+            Log.d("ZulipLog", "inside onNarrow1");
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Remove this logs

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -34,6 +34,7 @@
         android:label="@string/app_name"
         android:supportsRtl="false"
         android:theme="@style/AppTheme.DayNight.NoActionBar"
+        android:configChanges="orientation"
```
Comment (kunall17) [inline app/src/main/AndroidManifest.xml]: You must include screenSize also here, read the `Caution: Beginning with Android 3.2 (...` here https://developer.android.com/guide/topics/resources/runtime-changes.html

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 6]
```diff
@@ -301,6 +301,11 @@ protected void onCreate(Bundle savedInstanceState) {
         processParams();
 
         setContentView(R.layout.main);
+        populateViews();
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: 👍 for the change here

**Comment by divs4debu** [comment]
@kunall17  
- the toolbar hides on scroll and does not appears back i think thats the issue prevailing in the master branch.
- I was planning to do the layout changes i will do that today soon.
- No message is displayed that is another issue refer here  #481
I will complete these today the changes you requested.

**Comment by divs4debu** [comment]
@kunall17  Should commit all changes in new commit or squash them?

**Comment by divs4debu** [comment]
@kunall17 I have added new commit for your ease. Please review there are some prevailing error which are not due to these code i will open issues for that and everyone can collaborate and work on them.

**Comment by kunall17** [comment]
Squash the commits! 👍

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java (hunk 4 above)
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: This is left

**Comment by kunall17** [comment]
Change the commit message 
the search edit text is going out screen 
screen shot 2017-04-08 at 4 08 26 pm

The back icon still does not appears, and I have revert the update library, so those two issues are fixed!

**Comment by divs4debu** [comment]
@kunall17 Please Review

**Comment by kunall17** [comment]
@divs4debu there are merge conflicts, fix them as well!

**Comment by divs4debu** [comment]
@kunall17  Please review.

**Comment by kunall17** [comment]
- If in the homeView you should display a home icon on landscape, the back icon signifies that its narrowed currently
- there is a padding added top of search streams! 


screen shot 2017-04-12 at 10 22 43 pm

**Comment by divs4debu** [comment]
@kunall17 what is the significance of home icon in the home view?  Also providing bars icon would also be violation in master detail layout as the drawer is already open the only option I felt is back icon.  
Secondly I have by mistake added extra padding while correcting the merge conflicts.  I will verify and correct it again

**Comment by kunall17** [comment]
Yeah, so display no icon is in homeView and back icon if narrowed!

**Comment by divs4debu** [comment]
Ok thats better option of all

**Comment by divs4debu** [comment]
@kunall17  I have updated the pr. I need advice that there are some unnecessary checks for build version SDK 11 but our app is designed for minSDK 13 so should i remove them because that seems too clumsy.

**Comment by kunall17** [comment]
Yeah remove them!

**Comment by divs4debu** [comment]
@kunall17 Please review this commit then i will squash them.

**Comment by kunall17** [comment]
I tried it on 4.2.2 API 17, 1920x1200 320 dpi and toolbar is not showing up, and the padding is still there in the potrait mode! 
device-2017-04-16-152429


And on this device as well! 
screen shot 2017-04-16 at 3 37 15 pm

**Comment by divs4debu** [comment]
Ohk that might be the problem when I removed some checks.  Please can you tell what are the devices are you testing on I means give me all sdk versions and size and dpi.  I want make sure everything should work fine finally. !!

**Comment by kunall17** [comment]
@divs4debu I don't have any set of API which I'll be testing, I just picked up some lower versions than lollipop! 
I might give it a try on 16 as well! 

And it would be better to have separate commit for the lint issues, as I see you have edited some code which is not related to this PR.
Ideally you should selective squash the changes you have done related to this PR and rest in the separate commit, But its fine you can skip this step :) 👍

**Comment by divs4debu** [comment]
@kunall17  I think i should remove the redundant code as part of new issue because if it breaks any feature , that can easily be reverted and would be easy for You to look around the changes . This PR has already become too long due and I would like to implement this as fast as possible and create issue for redundant code and start fixing that. Is that ok?

**Comment by kunall17** [comment]
Your wish, it doesn't do anything big so even if the commit is added here in this PR, it would be fine for me + The travis will fail for this PR then it would be a problem as well (if you don't do selective squash)

**Comment by divs4debu** [comment]
@kunall17  please review

**Comment by kunall17** [comment]
I changed the commit message a bit and merged, thanks @divs4debu 👍

**Pull request closed without merging**